sqlflow/magic.py

```diff
@@ -8,27 +9,47 @@ class SqlFlowMagic(Magics):
     Provides the %%sqlflow magic
     """
 
-
     # TODO(tony): add config method such as
     # - default server url
     # - default credential
     # - default displaylimit
-    def __int__(self):
-        pass
+    def __init__(self, shell):
+        super(SqlFlowMagic, self).__init__(shell)
+        self.client = Client()
 
-    @line_magic('sqlflow')
-    def execute(self, line):
+    @cell_magic('sqlflow')
+    def execute(self, line, cell):
         """Runs SQL result against a sqlflow server, specified by server_url
 
         Example:
 
-            %%sqlflow localhost:50051
-            SELECT * FROM mytable
+            %%sqlflow  SELECT * FROM mytable
```

Comment:
Personally I like previous version which is much more flexible, otherwise the server url only can be set via environment variable.

**%%sqlflow localhost:50051
SELECT * FROM mytable**

Reply:
I think a client should be created during `%load_ext sqlflow.magic`, and a client issues a gRPC call on every `%%sqlflow STATEMENT`.

`%load_ext sqlflow.magic` doesn't take any arguments. So I used environment variables.

I agree that we can provide a feature where a user can do 

```python
%%sqlflow localhost:50051
SELECT * FROM mytable;
```

to create a client connecting to `localhost:50051`. This feature requires some parsing on the line string and cell string because a use can also write

```python
%%sqlflow SELECT * 
FROM mytable;
```

Reply:
Sounds good. How about

```python
%%sqlflow SELECT * # raise error "invalid configuration: SELECT *"
FROM mytable;

%%sqlflow localhost:50051 # overwrites self._client
SELECT * FROM mytable;
```